src/ip.rs

```diff
@@ -165,3 +168,50 @@ pub fn setup(lease: &NetavarkLease, interface: &str, ns_path: &str) -> Result<()
 pub fn teardown() -> Result<(), ProxyError> {
     todo!()
 }
+
+/// get_prefix_lengh takes a subnet mask in str form and
+/// returns its prefix length by counting ones.
+///
+/// # Arguments
+///
+/// * `netmask`: str form of subnet mask (i.e. 255.255.255.0)
+///
+/// returns: Result<u32, ProxyError>
+///
+/// # Examples
+///
+/// ```
+///
+/// ```
+fn get_prefix_length_v4(netmask: &str) -> Result<u32, ProxyError> {
+    let sub_mask = match Ipv4Addr::from_str(netmask) {
+        Ok(n) => n,
+        Err(e) => return Err(ProxyError::new(e.to_string())),
+    };
+    Ok(u32::from(sub_mask).count_ones())
+}
+
+#[cfg(test)]
```

Comment:
`netavark/aardvark` follows a format where a new file is created for tests instead of keeping with `src` implementation but its fine for now and maybe can be refactored later.